api/src/test/java/org/openmrs/module/fhir2/api/search/ServiceRequestSearchQueryImplTest.java

```diff
@@ -40,7 +102,565 @@ public void setup() throws Exception {
+		    referenceParam);
+		
+		IBundleProvider results = search(theParams);
+		
+		List<IBaseResource> resultList = get(results);
+		
+		assertThat(results, notNullValue());
+		assertThat(resultList, not(empty()));
+		assertThat(resultList.size(), greaterThanOrEqualTo(1));
+		assertThat(resultList,
+		    everyItem(hasProperty("subject", hasProperty("identifier", hasProperty("value", equalTo(PATIENT_IDENTIFIER))))));
+	}
+	
+	@Test
+	public void searchForTestOrders_shouldReturnEmptyListOfTestOrdersByMultiplePatientIdentifierAnd() {
+		ReferenceAndListParam referenceParam = new ReferenceAndListParam();
+		ReferenceParam patient = new ReferenceParam();
+		
+		patient.setValue(PATIENT_IDENTIFIER);
+		patient.setChain(Patient.SP_IDENTIFIER);
+		
+		ReferenceParam badPatient = new ReferenceParam();
+		
+		badPatient.setValue(PATIENT_WRONG_IDENTIFIER);
+		badPatient.setChain(Patient.SP_IDENTIFIER);
+		
+		referenceParam.addValue(new ReferenceOrListParam().add(patient)).addAnd(new ReferenceOrListParam().add(badPatient));
+		
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER,
+		    referenceParam);
+		IBundleProvider results = search(theParams);
+		
+		List<IBaseResource> resultList = get(results);
+		
+		assertThat(results, notNullValue());
+		assertThat(resultList, empty());
+	}
+	
+	@Test
+	public void searchForTestOrders_shouldReturnTestOrdersByEncounter() {
+		ReferenceAndListParam encounterReference = new ReferenceAndListParam()
+		        .addAnd(new ReferenceOrListParam().add(new ReferenceParam().setValue(ENCOUNTER_UUID)));
+		
+		SearchParameterMap theParams = new SearchParameterMap();
+		theParams.addParameter(FhirConstants.ENCOUNTER_REFERENCE_SEARCH_HANDLER, encounterReference);
+		
+		IBundleProvider results = search(theParams);
+		
+		assertThat(results, notNullValue());
+		
+		List<IBaseResource> resources = get(results);
+		
+		assertThat(resources, notNullValue());
+		assertThat(resources, hasItem(hasProperty("id", equalTo(TEST_ORDER_UUID))));
+	}
+	
+	@Test
+	public void searchForTestOrders_shouldReturnTestOrdersByPatientUuidAndPatientGivenName() {
+		
+		ReferenceAndListParam patientReference = new ReferenceAndListParam()
+		        .addAnd(new ReferenceOrListParam().add(new ReferenceParam().setValue(PATIENT_UUID)))
+		        .addAnd(new ReferenceOrListParam()
+		                .add(new ReferenceParam().setValue(PATIENT_GIVEN_NAME).setChain(Patient.SP_GIVEN)));
+		
+		SearchParameterMap theParams = new SearchParameterMap();
+		theParams.addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, patientReference);
+		
+		IBundleProvider results = search(theParams);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.size(), equalTo(10));
+		
+		List<IBaseResource> resources = get(results);
+		
+		assertThat(resources, not(empty()));
+		assertThat(resources.get(7).getIdElement().getIdPart(), equalTo(TEST_ORDER_UUID));
```

Comment:
It's better to write it as - 
`assertThat(resources, hasItem(hasProperty("id", equalTo(TEST_ORDER_UUID))));`

Please change this throughout the PR.